File: Java/problems/0044_maximum_average_subarray_i.java
//Problem Statement
//Given an array consisting of n integers, find the contiguous
//subarray of given length k that has the maximum average value.
//And you need to output the maximum average value.
class Main{
    public static void main(String[] args) {
        Solution sol = new Solution();
        int arr[] = {5,1,5,2,5,3,5,4};
        System.out.println(sol.findMaxAverage(arr,4));
    }
}

class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int start = 0;
        double result = Integer.MIN_VALUE, curr = 0;
        
        for(int i=0;i<nums.length;i++){
            if(i<k-1){
                curr+=nums[i];
            }else{
                curr+=nums[i];
                result = Math.max(result,curr/k);
                curr-=nums[start];
                start++;
            }
        }
        
        return result;
    }
}

File: Kotlin/src/main/kotlin/org/algorithm/problems/0044_maximum_average_subarray_i.kt
//Problem Statement
//Given an array consisting of n integers, find the contiguous
//subarray of given length k that has the maximum average value.
//And you need to output the maximum average value.
package org.algorithm.problems

class `0044_maximum_average_subarray_i` {
    fun findMaxAverage(nums: IntArray, k: Int): Double {
        var result: Double = Double.NEGATIVE_INFINITY;
        var actualSum: Double = 0.0;
        var start: Int = 0;
        var end: Int = 0;

        for (num in nums) {
            actualSum += num;
            end++;
            if (end - start == k) {
                var tempAvg: Double = 0.0;
                tempAvg = actualSum / k;
                result = maxOf(result, tempAvg);
                actualSum -= nums[start];
                start++;
            }
        }

        return result;
    }
}